config/passport.js

```diff
@@ -1,97 +1,35 @@
-const Users = require('../User').Users;
-const AuthTokenStrategy = require('passport-auth-token');
+const JwtStrategy = require('passport-jwt').Strategy
+const ExtractJwt = require('passport-jwt').ExtractJwt
+const keys = require('../config/keys')
+const Users = require('../User').Users
 
-module.exports = function (passport) {
-  passport.use(
-    'authtoken',
-    new AuthTokenStrategy(function (token, done) {
-      AccessToken.findOne(
-        {
-          where: {
-            id: token,
-          },
-        },
-        function (error, accessToken) {
-          if (error) {
-            console.log(1, error);
-            return done(error);
-          }
-
-          if (accessToken) {
-            if (!token.isValid(accessToken)) {
-              console.log(2);
-              return done(null, false);
-            }
 
-            Users.findOne(
-              {
-                where: {
-                  id: accessToken.id,
-                },
-              },
-              function (error, user) {
-                if (error) {
-                  console.log(3, error);
-                  return done(error);
-                }
+const options = {
+  jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(), 
+  secretOrKey: keys.jwt
+}
 
-                if (!user) {
-                  console.log(4);
-                  return done(null, false);
-                }
-                console.log(5);
-                return done(null, user);
-              },
-            );
-          } else {
-            return done(null);
-          }
-        },
-      );
-    }),
-  );
-};
-//const passport = require('passport');
-/*const LocalStrategy = require('passport-local').Strategy;
-const bcrypt = require('bcryptjs');
-const sequelize = require('../sequelize');
-const Users = require('../User').Users;
+module.exports = (passport) => {
 
-module.exports = function (passport) {
   passport.use(
-    new LocalStrategy({ usernameField: 'email' }, (email, password, done) => {
-      //Match User
-      Users.findOne({
-        where: {
-          email: email,
-        },
-      })
-        .then((user) => {
-          if (!user) {
-            return done(null, false, { message: 'That email is not regester' });
-          }
-
-          //Match password
-          bcrypt.compare(password, user.password, (err, isMatch) => {
-            if (err) throw err;
-            if (isMatch) {
-              return done(null, user);
-            } else {
-              return done(null, false, { message: 'Passport incorrect' });
-            }
-          });
-        })
-        .catch((err) => console.log(err));
-    }),
-  );
-
-  passport.serializeUser((user, done) => {
-    done(null, user.id);
-  });
+    new JwtStrategy(options, async(payload, done) => {
+      try{
+const user = await Users.findByPk( payload.userId ,{where:{
+email: 'email',
+id: 'id'
+}})
+
+if(user){
+  done(null, user)
+}else{
+  done(null, false)
```

Comment:
False - is incorrect response
I think will be better use `null `insted of `false`

Reply:
And what is the first argument?